emcache/protocol.py

```diff
@@ -255,10 +257,14 @@ async def create_protocol(
     else:
         ssl = False
 
+    if port:
```

Comment:
Im not really fan of using the port for configuring one or the other, unix sock address do not have a port. Ideally for supporting this we would need to create a new type here [1], which could be called `MemcachedHostUnixAddress` or something like that, and create a new union type which would be `MemcachedHostAddress` or `MemcachedHostUnixAddress`.

This will force us to change all signatures where we are specifying the type to the new generic one.

Not sure if this would break compatibility, we would need to check that no changes are required by current implementations


[1] https://github.com/emcache/emcache/blob/de148f5313217fe264bb950c36d75c18c515f501/emcache/node.py#L11